liota/lib/transports/mqtt.py

```diff
@@ -78,6 +79,12 @@ def on_disconnect(self, client, userdata, rc):
         self._connect_result_code = sys.maxsize
         self._disconnect_result_code = rc
         log.info("Disconnected with result code : {0} : {1} ".format(str(rc), paho.connack_string(rc)))
+        if self._disconnect_result_code == 0:
+            log.info("Clean disconnection")
+        else:
+            wait_time = round(random.uniform(1.0, 10.0), 1)
```

Comment:
Please use random.randint(1, 10)  as you do elsewhere, no need for floating point... and please add a comment explaining why we delay and how the upper bound of 10 seconds was chosen.

Reply:
I added the delay so that every mqtt client have the different random delay(different gateway) before retrying to connect, thus reducing the load on the broker. I used float trying to make it more random. Will add the comment.

Reply:
Being a float and rounding to a whole number doesn't make this any more random.

Reply:
I wasn't rounding to complete whole number it still had once decimal place but agreed not added much random effect.

Reply:
Python got me. :-) But I still disagree with using floats when they are not necessary.